rust/migrate-wicked/src/reader.rs

```diff
@@ -69,21 +97,32 @@ fn recurse_files(path: impl AsRef<Path>) -> std::io::Result<Vec<PathBuf>> {
     Ok(buf)
 }
 
-pub fn read(paths: Vec<String>) -> Result<Vec<Interface>, anyhow::Error> {
-    let mut interfaces: Vec<Interface> = vec![];
+pub fn read(paths: Vec<String>) -> Result<InterfacesResult, anyhow::Error> {
+    let mut result = InterfacesResult {
+        interfaces: vec![],
+        warning: None,
+    };
     for path in paths {
         let path: PathBuf = path.into();
         if path.is_dir() {
             let files = recurse_files(path)?;
             for file in files {
-                interfaces.append(&mut read_xml_file(file)?);
+                let mut read_xml = read_xml_file(file)?;
+                if result.warning.is_none() && read_xml.warning.is_some() {
+                    result.warning = read_xml.warning
```

Comment:
Do I oversee something, but maybe this is a reason why we should have a `Vec` in `InterfacesResult`, so the error of the first xml-file does not get overwritten by the error of the second file?

Reply:
Well in this case both warnings would be just the same. But maybe it would be nice to store the e.g. the interface names in the warning. I'll look into it.

Reply:
Mhm but then combining the warnings into one like we have to do in `migrate.rs` becomes annoying. I don't know, I think I'd like to keep it as is.

Reply:
I don't understand why the warnings are equal? Don't they show the unhandled field? And if the configurations differ, we would show differed fields?